addons/website/models/website.py

```diff
@@ -431,10 +512,39 @@ def set_images(images):
         }
         custom_resources = self._website_api_rpc('/api/website/1/configurator/custom_resources/%s' % kwargs.get('industry_id'), params)
 
+        # Update CTA
+        cta_data = website.get_cta_data(kwargs.get('website_purpose'))
+        if cta_data['cta_btn_text']:
+            xpath_views = ['website.snippets', 'website.layout']
+            for xpath_view in xpath_views:
+                parent_view = self.env['website'].with_context(website_id=website.id).viewref(xpath_view)
+                self.env['ir.ui.view'].create({
+                    'name': parent_view.key + ' CTA',
+                    'key': parent_view.key + "_cta",
+                    'inherit_id': parent_view.id,
+                    'website_id': website.id,
+                    'type': 'qweb',
+                    'priority': 32,
+                    'arch_db': """
+                        <data>
+                            <xpath expr="//t[@t-set='cta_btn_href']" position="replace">
+                                <t t-set="cta_btn_href">%s</t>
+                            </xpath>
+                            <xpath expr="//t[@t-set='cta_btn_text']" position="replace">
+                                <t t-set="cta_btn_text">%s</t>
+                            </xpath>
+                            <xpath expr="//t[@t-set='cta_btn_text_uppercase']" position="replace">
+                                <t t-set="cta_btn_text_uppercase">%s</t>
+                            </xpath>
+                        </data>
+                    """ % (cta_data['cta_btn_href'], cta_data['cta_btn_text'], cta_data['cta_btn_text_uppercase'])
+                })
+        self.env['website.menu'].search([('website_id', '=', website.id), ('url', '=', cta_data['cta_btn_href'])]).unlink()
```

Comment:
This line implements the spec: "remove menu link 'Contact us' when CTA is 'contact us' and 'Appointment' link when CTA is Schedule an appointment".  But not ok since some menu do not have CTA button...